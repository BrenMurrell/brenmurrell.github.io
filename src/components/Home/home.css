/* :root {
    --pulse-speed: 8s;
} */

@keyframes fadeIn {
    0% {
        opacity: 0;
    }
    15% {
        opacity: 1;
    }
    100% {
        opacity: 1;
    }
}

@keyframes pulse {
    0% {
        opacity: 0;
    }
    15% {
        opacity: 1;
    }
    30% {
        opacity: 1;
    }
    35% {
        opacity: 0;
    }
    100% {
        opacity: 0;
    }
}

.home {
    display: flex;
    align-items: center;
    justify-content: center;
    min-height: calc(100vh - 5rem);
    /* width: calc(100% - 3rem); */
}

.home__content {
    position: relative;
    min-width: 22rem;
}

.home h1 {
    margin-bottom: 3rem;
}

.home h2 {
    position: absolute;
    bottom: 0;
    left: 0;
    opacity: 0;
    animation: pulse var(--pulse-speed) forwards;
    animation-delay: 0;
}
.home h2:nth-of-type(2) {
    animation-delay: calc(var(--pulse-speed)/ 3);
}

.home h2:nth-of-type(3) {
    animation-delay: calc(var(--pulse-speed)/ 3 * 2);
}
.home h2:nth-of-type(4) {
    animation: fadeIn var(--pulse-speed) forwards;
    animation-delay: calc(var(--pulse-speed));
}

.home h2:nth-of-type(4):after {
    content: '';
    display: block;
    height: 0.25rem;
    width: 0%;
    position: absolute;
    bottom: 0;
    left: 0;
    background-color: var(--fg);
    animation: stretch calc(var(--pulse-speed) / 6) forwards;
    animation-delay: var(--pulse-speed);
}

@keyframes stretch {
    0% {
        width: 0;
    }
    100% {
        width: 100%;
    }
}

.home h2:after {
    content: '...';
}