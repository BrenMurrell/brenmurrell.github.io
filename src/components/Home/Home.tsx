import React from 'react';
import './home.css';

const Home: React.FC = () => {
    return (
        <div className="home">
            <div className="home__content">
                <h1>Bren Murrell</h1>
                <h2>Senior Front-End Developer</h2>
                <h2>Accessibility Advocate</h2>
                <h2>Grower of people</h2>
                <h2>Your next hire.</h2>
            </div>
        </div>
    );
};

export default Home;