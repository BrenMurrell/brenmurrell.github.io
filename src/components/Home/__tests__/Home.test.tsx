import { render, screen } from "@testing-library/react";
import Home from "../Home";

describe('Home', () => {
    // it.todo('fix tests');
    it('renders' ,() => {
        const container = render(<Home />);
        expect(container).toMatchSnapshot();
    })

    it('has correct content', () => {
        render(<Home />);
        const heading_one = screen.getByRole('heading', { level: 1});
        expect(heading_one.textContent).toEqual("Bren Murrell");

        const subheadings = screen.queryAllByRole('heading', { level: 2});
        expect(subheadings).toHaveLength(4);
        expect(subheadings[0].textContent).toEqual('Senior Front End Developer');
    })
});