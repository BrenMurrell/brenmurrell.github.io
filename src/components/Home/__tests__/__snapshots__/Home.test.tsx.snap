// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home"
      >
        <div
          class="home__content"
        >
          <h1>
            Bren Murrell
          </h1>
          <h2>
            Senior Front-End Developer
          </h2>
          <h2>
            Accessibility Advocate
          </h2>
          <h2>
            Grower of people
          </h2>
          <h2>
            Your next hire.
          </h2>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home"
    >
      <div
        class="home__content"
      >
        <h1>
          Bren Murrell
        </h1>
        <h2>
          Senior Front-End Developer
        </h2>
        <h2>
          Accessibility Advocate
        </h2>
        <h2>
          Grower of people
        </h2>
        <h2>
          Your next hire.
        </h2>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
